liquibase-core/src/main/java/liquibase/database/core/PostgresDatabase.java

```diff
@@ -297,7 +297,8 @@ public String correctObjectName(String objectName, Class<? extends DatabaseObjec
         if (objectName.contains("-")
                 || hasMixedCase(objectName)
                 || startsWithNumeric(objectName)
-                || isReservedWord(objectName)) {
+                || isReservedWord(objectName)
+                || Boolean.TRUE.equals(GlobalConfiguration.PRESERVE_SCHEMA_CASE.getCurrentValue())) {
```

Comment:
This will make it so that postgresql preserve the case on ALL objects if preserveSchemaCase is set?

I think you need to take this out of this if block and put it as a separate if block checking the objectType as the first lines of this function

Reply:
Okay, I see your point.  Testing didn't find any issues, but maybe they just didn't hit one.

Reply:
I wouldn't have seen any issues in my manual testing because my non-schema objects were all lowercase for the Postgres tests. Even in the automated functional tests, we generally use the default casing for any given database platform unless we are specifically testing casing. If there is/was an issue, I can see why test would _not_ have seen it.